lib/decidim/proposals/admin_engine.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Proposals
+    # This is the engine that runs on the public interface of `decidim-proposals`.
+    class AdminEngine < ::Rails::Engine
+      isolate_namespace Decidim::Proposals::Admin
+
+      paths["db/migrate"] = nil
+      paths["lib/tasks"] = nil
+
+      routes do
+        resources :proposals, only: [:show, :index, :new, :create, :edit, :update] do
+          member do
+            put :hide
+          end
+          resources :valuation_assignments, only: [:destroy]
+          collection do
+            post :update_category
+            post :publish_answers
+            post :update_scope
+            resource :proposals_import, only: [:new, :create]
+            resource :proposals_merge, only: [:create]
+            resource :proposals_split, only: [:create]
+            resource :valuation_assignment, only: [:create, :destroy]
+          end
+          resources :proposal_answers, only: [:edit, :update]
+          resources :proposal_notes, only: [:create]
+        end
+
+        resources :participatory_texts, only: [:index] do
+          collection do
+            get :new_import
+            post :import
+            patch :import
+            post :update
+            post :discard
+          end
+        end
+
+        root to: "proposals#index"
+      end
+
+      def load_seed
+        nil
+      end
+    end
+  end
+end
```

Comment:
```suggestion
```
I don't like this override, we don't really need it in order to just add an action to the admin proposals controller.
Overriding all of this will make this module too unstable.

I would prefer to create a new controller with our "hide" action in it. Not to override the Admin::ProposalsController.

Then we will have to create a new `lib/decidim/reporting_proposals/admin_engine.rb` with this route.